django/sierra/sierra/routers.py

```diff
@@ -0,0 +1,41 @@
+from django.conf import settings
+from django.db import DatabaseError
+
+
+class SierraRouter(object):
+    """
+    The DB router used by the Catalog API project.
+    """
+    def db_for_read(self, model, **hints):
+        """
+        Routes all read attempts for base models to Sierra DB.
+        """
+        return 'sierra' if model._meta.app_label == 'base' else 'default'
+
+    def db_for_write(self, model, **hints):
+        """
+        Raises an error for write attempts to Sierra DB, and routes
+        all others to default. Note that base.models have their save
+        and delete methods disabled, so attempting to write base.models
+        should never call this code in the first place.
+        """
+        if model._meta.app_label == 'base':
+            raise DatabaseError('Attempted to write to Sierra database.')
+        else:
+            return 'default'
+
+    def allow_migrate(self, db, model):
+        """
+        Ensures that, 1. sierra migrations happen when running tests,
+        2. sierra migrations don't happen when not running tests,
+        3. sierra data (everything in base.models) only goes into the
+        sierra test database, and 4. non-sierra data only goes into
+        the default database.
+        """
+        allow = None
+        if model._meta.app_label == 'base':
+            testing = db == 'sierra' and settings.TESTING
+            allow = True if testing else False
+        else:
+            allow = True if db != 'sierra' else False
+        return allow
```

Comment:
One last thought:

There are differing opinions on having multiple return statements in a functions. I think there is a time and place for both.

I think it is fine the way it currently is written, to return whatever has been assigned to the var `allow`. 

I think it would also be fine to return directly from the if/else block like so

```python
def allow_migrate(self, db, model):
    if model._meta.app_label == 'base':
       return  db == 'sierra' and settings.TESTING
    else:
       return db != 'sierra'
```

Just throwing that out there :smile: